monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+https://github.com/Project-MONAI/MONAI/wiki/MONAI_Design
+"""
+
+from typing import Any, List, Optional
+
+import numpy as np
+
+from monai.transforms.transform import RandomizableTransform, Transform
+from monai.utils import optional_import
+from monai.utils.enums import TransformBackends
+
+zoom, has_zoom = optional_import("scipy.ndimage", name="zoom")
+resample_poly, has_resample_poly = optional_import("scipy.signal", name="resample_poly")
+fft, has_resample_fft = optional_import("scipy.signal", name="resample")
+shift, has_shift = optional_import("scipy.ndimage.interpolation", name="shift")
+square, has_square = optional_import("scipy.signal", name="square")
+iirnotch, has_iirnotch = optional_import("scipy.signal", name="iirnotch")
+filtfilt, has_filtfilt = optional_import("scipy.signal", name="filtfilt")
+spectrogram, has_spectrogram = optional_import("scipy.signal", name="spectrogram")
+central_frequency, has_central_frequency = optional_import("pywt", name="central_frequency")
+cwt, has_cwt = optional_import("pywt", name="cwt")
+
+
+__all__ = [
+    "SignalRandDrop",
+    "SignalRandScale",
+    "SignalRandShift",
+    "SignalResample",
+    "SignalRandAddSine",
+    "SignalRandAddSquarePulse",
+    "SignalRandAddGaussianNoise",
+    "SignalRandAddSinePartial",
+    "SignalRandAddSquarePulsePartial",
+    "SignalNormalize",
+    "SignalStandardize",
+    "SignalZeroPad",
+    "SignalFillEmpty",
+    "SignalRemoveFrequency",
+    "SignalShortTimeFourier",
+    "SignalContinousWavelet",
+]
+
+
+class SignalResample(Transform):
+    """
+    Resample signal to the target sampling rate.
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self, method: str = "interpolation", current_sample_rate: int = 500, target_sample_rate: int = 250
+    ) -> None:
+        """
+        Args:
+            method: which method to be used to resample the signal, default is interpolation.
+            Available options : {``"interpolation"``, ``"polynomial"``, ``"fourier"``}
+            current_sample_rate: initial sampling rate of the signal
+            target_sample_rate: target signal sample rate
+        """
+        if method is None and method not in ["interpolation", "polynomial", "fourier"]:
+            raise ValueError("Incompatible values: method needs to be either interpolation, polynomial or fourier.")
+        if current_sample_rate < target_sample_rate:
+            raise ValueError(
+                "Incompatible target_sampling_rate: target_sampling_rate must be lower current_sampling_rate."
+            )
+        self.method = method
+        self.current_sample_rate = current_sample_rate
+        self.target_sample_rate = target_sample_rate
+
+    def __call__(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Args:
+            signal: input 1 dimension signal to be resampled
+        """
+        if len(signal.shape) > 1:
+            inputs_channels = signal.shape[0]
```

Comment:
All of our transforms assume the channel dimension to be present, so maybe these transforms should make the same assumption to be consistent with them.

Reply:
fair enough

Reply:
removed , assumed channel is already there